nexus-repository-p2-it/src/test/java/org/sonatype/nexus/repository/p2/internal/P2ProxyIT.java

```diff
@@ -0,0 +1,158 @@
+
+  private static final String PACKAGE_BASE_PATH = "R-4.7.3a-201803300640/features/";
+
+  private static final String BAD_PATH = "/this/path/is/not/valid";
+
+  private static final String VALID_PACKAGE_URL = PACKAGE_BASE_PATH + PACKAGE_NAME;
+
+  private static final String INVALID_PACKAGE_URL = PACKAGE_BASE_PATH + INVALID_PACKAGE_NAME;
+
+  private P2Client proxyClient;
+
+  private Repository proxyRepo;
+
+  private Server server;
+
+  @Configuration
+  public static Option[] configureNexus() {
+    return NexusPaxExamSupport.options(
+        NexusITSupport.configureNexusBase(),
+        nexusFeature("org.sonatype.nexus.plugins", "nexus-repository-p2")
+    );
+  }
+
+  @Before
+  public void setup() throws Exception {
+    server = Server.withPort(0)
+        .serve("/" + VALID_PACKAGE_URL)
+        .withBehaviours(Behaviours.file(testData.resolveFile(PACKAGE_NAME)))
+        .serve("/" + ARTIFACT_JAR)
+        .withBehaviours(Behaviours.file(testData.resolveFile(ARTIFACT_JAR)))
+        .serve("/" + ARTIFACT_XML)
+        .withBehaviours(Behaviours.file(testData.resolveFile(ARTIFACT_XML)))
+        .serve("/" + ARTIFACT_XML_XZ)
+        .withBehaviours(Behaviours.file(testData.resolveFile(ARTIFACT_XML_XZ)))
+        .start();
+
+    proxyRepo = repos.createP2Proxy("p2-test-proxy", server.getUrl().toExternalForm());
+    proxyClient = p2Client(proxyRepo);
+  }
+
+  @Test
+  public void unresponsiveRemoteProduces404() throws Exception {
+    assertThat(status(proxyClient.get(BAD_PATH)), is(HttpStatus.NOT_FOUND));
+  }
+
+  @Test
+  public void retrieveJarFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(VALID_PACKAGE_URL)), is(HttpStatus.OK));
+
+    final Asset asset = findAsset(proxyRepo, "/" + VALID_PACKAGE_URL);
+    assertThat(asset.name(), is(equalTo("/" + VALID_PACKAGE_URL)));
+    assertThat(asset.contentType(), is(equalTo(MIME_TYPE)));
+    assertThat(asset.format(), is(equalTo(FORMAT_NAME)));
+
+    final Component component = findComponent(proxyRepo, COMPONENT_NAME);
+    assertThat(component.version(), is(equalTo(VERSION_NUMBER)));
+    assertThat(component.group(), is(equalTo(null)));
+  }
+
+  @Test
+  public void retrieveZipFromProxyShouldNotWork() throws Exception {
+    assertThat(status(proxyClient.get(INVALID_PACKAGE_URL)), is(HttpStatus.NOT_FOUND));
+  }
+
+  @Test
+  public void removeMirrorURLFromXML() throws Exception {
+    assertThat(status(proxyClient.get(ARTIFACT_XML)), is(HttpStatus.OK));
+
+    try (CloseableHttpResponse response = proxyClient.get(ARTIFACT_XML)) {
+      HttpEntity entity = response.getEntity();
+      String result = IOUtils.toString(entity.getContent()).replaceAll("\\s+","");
+
+      InputStream targetStream = new FileInputStream(testData.resolveFile(ARTIFACT_WITHOUT_MIRROR_XML));
+      String expected = IOUtils.toString(targetStream).replaceAll("\\s+","");
+
+      assertThat(result, equalTo(expected));
+
```

Comment:
IMHO: extra line

Reply:
in: https://github.com/sonatype-nexus-community/nexus-repository-p2/pull/17/commits/94912ebe5ebec6d38ce4e6c99a076d6a7b89a042